app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb

```diff
@@ -0,0 +1,41 @@
+module ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::EventParser
+  def self.event_to_hash(event, ems_id)
+    event_hash = {
+      :event_type => "#{event[:resource]}.#{event[:action]}",
+      :source     => "IBMCloud-PowerVS",
+      :ems_id     => ems_id,
+      :ems_ref    => event[:eventID],
+      :timestamp  => event[:time],
+      :full_data  => event,
+      :username   => event[:user][:email]
+    }
+
+    _log.debug("event_hash=#{event_hash}")
+    case event_hash[:event_type]
+    when /^pvm-instance/
+      parse_vm_event!(event, event_hash)
+    end
+
+    event_hash
+  end
+
+  def self.parse_vm_event!(event, event_hash)
+    pvm_instance_name = event[:message].split('\'')[1]
+    ems = ExtManagementSystem.find(event_hash[:ems_id])
+    vm = ems.vms.find_by(:name => pvm_instance_name)
+
+    if vm.nil?
+      if ['update', 'create'].include?(event['type'])
```

Comment:
Could we do this check before doing the query to find the vm?  The event catcher side is really intended to not run any DB queries to maintain high event throughput, that is why all of the queries are done by the event_handler once the events have been caught.

Reply:
[done](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/279/commits/572010c76a2f29bf7681667c5bad8bdb1fb28b85#diff-94f8045f8f587fbabedbe1ade93e3282f34b17895fd96a490f4a095aa5d697edR23)